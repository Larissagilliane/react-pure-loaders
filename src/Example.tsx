import * as React from 'react';

export const Example = (props: any) => {
  return <span {...props}>Hello World!</span>;
};