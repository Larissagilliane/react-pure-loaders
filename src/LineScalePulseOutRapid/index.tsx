import { css } from 'glamor';
import * as React from 'react';
import style from './style';

export interface IGeneralPropTypes {
  loading: boolean;
}

export class LineScalePulseOutRapid extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { loading } = this.props;

    if (loading) {
      return (
        <div {...css(style)}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
