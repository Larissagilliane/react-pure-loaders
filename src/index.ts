/**
 * Dots
 */
export { BallBeat } from './BallBeat';
export { BallClipRotate } from './BallClipRotate';
export { BallClipRotateMultiple } from './BallClipRotateMultiple';
export { BallClipRotatePulse } from './BallClipRotatePulse';
export { BallGridBeat } from './BallGridBeat';
export { BallGridPulse } from './BallGridPulse';
export { BallPulse } from './BallPulse';
export { BallPulseRise } from './BallPulseRise';
export { BallPulseRound } from './BallPulseRound';
export { BallPulseSync } from './BallPulseSync';
export { BallRotate } from './BallRotate';
export { BallScale } from './BallScale';
export { BallScaleMultiple } from './BallScaleMultiple';
export { BallScaleRandom } from './BallScaleRandom';
export { BallScaleRipple } from './BallScaleRipple';
export { BallScaleRippleMultiple } from './BallScaleRippleMultiple';
export { BallSpinFadeLoader } from './BallSpinFadeLoader';
export { BallSpinLoader } from './BallSpinLoader';
export { BallTrianglePath } from './BallTrianglePath';
export { BallZigZag } from './BallZigZag';
export { BallZigZagDeflect } from './BallZigZagDeflect';

/**
 * Lines
 */
export { LineScale } from './LineScale';
export { LineScaleParty } from './LineScaleParty';
export { LineScalePulseOut } from './LineScalePulseOut';
export { LineScalePulseOutRapid } from './LineScalePulseOutRapid';
export { LineScaleRandom } from './LineScaleRandom';
export { LineSpinFadeLoader } from './LineSpinFadeLoader';

/**
 * Misc
 */
export { CubeTransition } from './CubeTransition';
export { Pacman } from './Pacman';
export { SemiCircleSpin } from './SemiCircleSpin';
export { SquareSpin } from './SquareSpin';
export { TriangleSkewSpin } from './TriangleSkewSpin';
