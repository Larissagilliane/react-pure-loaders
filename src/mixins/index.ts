import { css } from 'glamor';
import {
  BALL_SIZE,
  LINE_HEIGHT,
  LINE_WIDTH,
  MARGIN,
  PRIMARY_COLOR,
} from '../variables';

interface IGrid {
  [id: string]: object;
}

export function globalBg() {
  return {
    backgroundColor: PRIMARY_COLOR,
  };
}

export function globalAnimation() {
  return {
    animationFillMode: 'both',
  };
}

export function balls() {
  return css(
    {
      borderRadius: '100%',
      height: BALL_SIZE,
      margin: MARGIN,
      width: BALL_SIZE,
    },
    globalBg(),
  );
}

export function ballGrid() {
  let i = 1;
  const grid: IGrid = {};
  for (; i < 10; i += 1) {
    const delay = ((parseInt((Math.random() * 100).toString(), 10) / 100) - 0.2);
    const duration = ((parseInt((Math.random() * 100).toString(), 10) / 100) + 0.6);

    grid[` > div:nth-child(${i})`] = {
      animationDelay: `${delay}s`,
      animationDuration: `${duration}s`,
    };
  }

  return grid;
}

export function lines() {
  return css(
    {
      borderRadius: '2px',
      height: LINE_HEIGHT,
      margin: MARGIN,
      width: LINE_WIDTH,
    },
    globalBg(),
  );
}
