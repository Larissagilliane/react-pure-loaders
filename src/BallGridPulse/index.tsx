import { css } from 'glamor';
import * as React from 'react';
import { IGeneralPropTypes } from '../mixins';
import style from './style';

export class BallGridPulse extends React.PureComponent<IGeneralPropTypes, {}> {

  public render() {
    const { color, loading } = this.props;

    if (loading) {
      return (
        <div {...css(style(color))}>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
          <div/>
        </div>
      );
    }

    return null;
  }
}
