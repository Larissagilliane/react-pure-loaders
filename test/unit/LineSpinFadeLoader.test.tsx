import { mount, render, shallow } from 'enzyme';
import { StyleSheet } from 'glamor/lib/sheet';
import * as React from 'react';
import { Pacman } from '../../src';

const stylesheet = new StyleSheet();

describe('<Pacman>', () => {
  beforeEach(() => {
    stylesheet.flush();
  });

  test('<Pacman> Shallow', () => {
    const ui = (<Pacman loading={true}/>);

    expect(shallow(ui)).toMatchSnapshot(`enzyme.shallow`);
  });
  test('<Pacman> Mount', () => {
    const ui = (<Pacman loading={true}/>);

    expect(mount(ui)).toMatchSnapshot(`enzyme.mount`);
  });
  test('<Pacman> Render', () => {
    const ui = (<Pacman loading={true}/>);

    expect(render(ui)).toMatchSnapshot(`enzyme.render`);
  });
});
