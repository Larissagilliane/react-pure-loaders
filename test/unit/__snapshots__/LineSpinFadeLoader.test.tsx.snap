// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pacman> <Pacman> Mount: enzyme.mount 1`] = `
.css-11jr1oo,
[data-css-11jr1oo] {
  position: relative;
}

.css-11jr1oo> div:nth-child(2),
[data-css-11jr1oo]> div:nth-child(2) {
  animation: animation_1w5vsxu 1s -0.99s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.99s infinite linear;
}

.css-11jr1oo> div:nth-child(3),
[data-css-11jr1oo]> div:nth-child(3) {
  animation: animation_1w5vsxu 1s -0.66s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.66s infinite linear;
}

.css-11jr1oo> div:nth-child(4),
[data-css-11jr1oo]> div:nth-child(4) {
  animation: animation_1w5vsxu 1s -0.33s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.33s infinite linear;
}

.css-11jr1oo> div:nth-child(5),
[data-css-11jr1oo]> div:nth-child(5) {
  animation: animation_1w5vsxu 1s 0.00s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s 0.00s infinite linear;
}

.css-11jr1oo > div:first-of-type,
[data-css-11jr1oo] > div:first-of-type {
  border-bottom: 25px solid #ffffff;
  border-left: 25px solid #ffffff;
  border-radius: 25px;
  border-right: 25px solid transparent;
  border-top: 25px solid #ffffff;
  height: 0px;
  width: 0px;
  animation: animation_1dfnslg 0.5s 0s infinite;
  left: -30px;
  position: relative;
  -webkit-animation: animation_1dfnslg 0.5s 0s infinite;
}

.css-11jr1oo > div:nth-child(2),
[data-css-11jr1oo] > div:nth-child(2) {
  border-bottom: 25px solid #ffffff;
  border-left: 25px solid #ffffff;
  border-radius: 25px;
  border-right: 25px solid transparent;
  border-top: 25px solid #ffffff;
  height: 0px;
  width: 0px;
  animation: animation_mvibk4 0.5s 0s infinite;
  left: -30px;
  margin-top: -50px;
  position: relative;
  -webkit-animation: animation_mvibk4 0.5s 0s infinite;
}

.css-11jr1oo > div:nth-child(3),
[data-css-11jr1oo] > div:nth-child(3),
.css-11jr1oo > div:nth-child(4),
[data-css-11jr1oo] > div:nth-child(4),
.css-11jr1oo > div:nth-child(5),
[data-css-11jr1oo] > div:nth-child(5),
.css-11jr1oo > div:nth-child(6),
[data-css-11jr1oo] > div:nth-child(6) {
  border-radius: 100%;
  height: 10px;
  margin: 2px;
  width: 10px;
  background-color: #ffffff;
  left: 70px;
  position: absolute;
  top: 25px;
  transform: translate(0,-6.25px);
  -webkit-transform: translate(0,-6.25px);
}

<Pacman
  loading={true}
>
  <div
    data-css-11jr1oo=""
  >
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</Pacman>
`;

exports[`<Pacman> <Pacman> Render: enzyme.render 1`] = `
.css-11jr1oo,
[data-css-11jr1oo] {
  position: relative;
}

.css-11jr1oo> div:nth-child(2),
[data-css-11jr1oo]> div:nth-child(2) {
  animation: animation_1w5vsxu 1s -0.99s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.99s infinite linear;
}

.css-11jr1oo> div:nth-child(3),
[data-css-11jr1oo]> div:nth-child(3) {
  animation: animation_1w5vsxu 1s -0.66s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.66s infinite linear;
}

.css-11jr1oo> div:nth-child(4),
[data-css-11jr1oo]> div:nth-child(4) {
  animation: animation_1w5vsxu 1s -0.33s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.33s infinite linear;
}

.css-11jr1oo> div:nth-child(5),
[data-css-11jr1oo]> div:nth-child(5) {
  animation: animation_1w5vsxu 1s 0.00s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s 0.00s infinite linear;
}

.css-11jr1oo > div:first-of-type,
[data-css-11jr1oo] > div:first-of-type {
  border-bottom: 25px solid #ffffff;
  border-left: 25px solid #ffffff;
  border-radius: 25px;
  border-right: 25px solid transparent;
  border-top: 25px solid #ffffff;
  height: 0px;
  width: 0px;
  animation: animation_1dfnslg 0.5s 0s infinite;
  left: -30px;
  position: relative;
  -webkit-animation: animation_1dfnslg 0.5s 0s infinite;
}

.css-11jr1oo > div:nth-child(2),
[data-css-11jr1oo] > div:nth-child(2) {
  border-bottom: 25px solid #ffffff;
  border-left: 25px solid #ffffff;
  border-radius: 25px;
  border-right: 25px solid transparent;
  border-top: 25px solid #ffffff;
  height: 0px;
  width: 0px;
  animation: animation_mvibk4 0.5s 0s infinite;
  left: -30px;
  margin-top: -50px;
  position: relative;
  -webkit-animation: animation_mvibk4 0.5s 0s infinite;
}

.css-11jr1oo > div:nth-child(3),
[data-css-11jr1oo] > div:nth-child(3),
.css-11jr1oo > div:nth-child(4),
[data-css-11jr1oo] > div:nth-child(4),
.css-11jr1oo > div:nth-child(5),
[data-css-11jr1oo] > div:nth-child(5),
.css-11jr1oo > div:nth-child(6),
[data-css-11jr1oo] > div:nth-child(6) {
  border-radius: 100%;
  height: 10px;
  margin: 2px;
  width: 10px;
  background-color: #ffffff;
  left: 70px;
  position: absolute;
  top: 25px;
  transform: translate(0,-6.25px);
  -webkit-transform: translate(0,-6.25px);
}

<div
  data-css-11jr1oo=""
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`<Pacman> <Pacman> Shallow: enzyme.shallow 1`] = `
.css-11jr1oo,
[data-css-11jr1oo] {
  position: relative;
}

.css-11jr1oo> div:nth-child(2),
[data-css-11jr1oo]> div:nth-child(2) {
  animation: animation_1w5vsxu 1s -0.99s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.99s infinite linear;
}

.css-11jr1oo> div:nth-child(3),
[data-css-11jr1oo]> div:nth-child(3) {
  animation: animation_1w5vsxu 1s -0.66s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.66s infinite linear;
}

.css-11jr1oo> div:nth-child(4),
[data-css-11jr1oo]> div:nth-child(4) {
  animation: animation_1w5vsxu 1s -0.33s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s -0.33s infinite linear;
}

.css-11jr1oo> div:nth-child(5),
[data-css-11jr1oo]> div:nth-child(5) {
  animation: animation_1w5vsxu 1s 0.00s infinite linear;
  -webkit-animation: animation_1w5vsxu 1s 0.00s infinite linear;
}

.css-11jr1oo > div:first-of-type,
[data-css-11jr1oo] > div:first-of-type {
  border-bottom: 25px solid #ffffff;
  border-left: 25px solid #ffffff;
  border-radius: 25px;
  border-right: 25px solid transparent;
  border-top: 25px solid #ffffff;
  height: 0px;
  width: 0px;
  animation: animation_1dfnslg 0.5s 0s infinite;
  left: -30px;
  position: relative;
  -webkit-animation: animation_1dfnslg 0.5s 0s infinite;
}

.css-11jr1oo > div:nth-child(2),
[data-css-11jr1oo] > div:nth-child(2) {
  border-bottom: 25px solid #ffffff;
  border-left: 25px solid #ffffff;
  border-radius: 25px;
  border-right: 25px solid transparent;
  border-top: 25px solid #ffffff;
  height: 0px;
  width: 0px;
  animation: animation_mvibk4 0.5s 0s infinite;
  left: -30px;
  margin-top: -50px;
  position: relative;
  -webkit-animation: animation_mvibk4 0.5s 0s infinite;
}

.css-11jr1oo > div:nth-child(3),
[data-css-11jr1oo] > div:nth-child(3),
.css-11jr1oo > div:nth-child(4),
[data-css-11jr1oo] > div:nth-child(4),
.css-11jr1oo > div:nth-child(5),
[data-css-11jr1oo] > div:nth-child(5),
.css-11jr1oo > div:nth-child(6),
[data-css-11jr1oo] > div:nth-child(6) {
  border-radius: 100%;
  height: 10px;
  margin: 2px;
  width: 10px;
  background-color: #ffffff;
  left: 70px;
  position: absolute;
  top: 25px;
  transform: translate(0,-6.25px);
  -webkit-transform: translate(0,-6.25px);
}

<div
  data-css-11jr1oo=""
>
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
